packages/create-styles/src/createStyleSystem/utils.js

```diff
@@ -127,3 +127,32 @@ export function getDisplayName(Component) {
 
 	return displayName;
 }
+
+/**
+ * Resolves and compiles interpolated CSS styles for styled-components.
+ * Allows for prop (function) interpolation within the style rules.
+ *
+ * For more information on tagged template literals, check out:
+ * https://mxstbr.blog/2016/11/styled-components-magic-explained/
+ *
+ * @param {Array<string,function>} interpolatedStyles The interpolated styles from a Styled component.
+ * @param {Object} props Incoming component props.
+ * @returns {string} Compiled CSS style rules.
+ */
+export function compileInterpolatedStyles(interpolatedStyles, props) {
+	const resolvedInterpolatedStyles = interpolatedStyles.map((a) =>
+		is.function(a) ? a(props) : a,
+	);
+
+	const [cssChunk, ...interpolatedCSSRule] = resolvedInterpolatedStyles;
+
+	const rules = [...cssChunk].slice(0, -1);
+	const end = [...cssChunk].slice(-1);
+
+	const compiledStyles = rules
+		.map((rule, index) => rule + interpolatedCSSRule[index])
+		.join('')
+		.concat(end);
```

Comment:
Can you briefly describe what's going on here? I'm having a hard time grokking it.

Reply:
@saramarcondes To be honest... I'm not sure 😂 . It was code [I've written before and borrowed](https://github.com/helpscout/fancy/blob/1.1.0/src/utilities/primitives.js#L39) for this update.

For context, the code I've written before needed to do the same CSS-in-JS interpolating thing.

---

I **believe** it's something like this!

Incoming, the `interpolatedStyles` may look like this example:

```
interpolatedStyles = [['a', 'b', ';'], fn, fn]
```

`resolvedInterpolatedStyles` resolves the functions (`fn`), so it becomes this:


```
resolvedInterpolatedStyles = [['a', 'b', ';'], 'one', 'two']
```

Next, we destructure:

```
const [cssChunk, ...interpolatedCSSRule] = resolvedInterpolatedStyles;
// cssChunk = ['a', 'b', ';']
// interpolatedCSSRule = ['one', 'two']
```

Then, we split off `rules` and `end`:

```
rules = ['a', 'b']
end = [';']
```

Lastly, we combine it all together in that `.map` + `.join`, so that we get:

```
aonebtwo;
```

(The actual style rules would have `;` in between the current chunks.)

---

Hope this helps!